.github/workflows/beam_PreCommit_Spotless.yml

```
@@ -94,4 +94,9 @@ jobs:
         uses: actions/upload-artifact@v3
         with:
           name: java-code-coverage-report
-          path: "**/build/reports/checkstyle/*.xml"
\ No newline at end of file
+          path: "**/build/reports/checkstyle/*.xml"
+      - name: Publish checkstyle check
+        uses: jwgmeligmeyling/checkstyle-github-action@master
```

```suggestion
        uses: jwgmeligmeyling/checkstyle-github-action@v1
```

Generally when we use 3rd party actions, we should pin to a tag. This is less likely to randomly break, and the infra team will only allowlist actions that are pinned to tags

---

I created a jira ticket with infra (may not be publicly visible) to allow list this action @ v1 https://issues.apache.org/jira/browse/INFRA-24948

---

Okay, thank you!

---

The ticket was approved, merging